src/pages/ProductsListing/ProductsListing.js

```diff
@@ -1,24 +1,41 @@
 import { Filter, Footer, Navbar, ProductsCard } from "../../components";
-import { useProducts } from "../../contexts/products-context";
+import { useProducts, useSortFilter } from "../../contexts";
+import {
+  sortFunc,
+  categoryFilterFunc,
+  priceSliderFunc,
+  ratingFilterFunc,
+} from "../../helpers";
 import "./ProductsListing.css";
 
 const ProductsListing = () => {
   const { products: availableProducts } = useProducts();
+  const {
+    state: { sortBy, categories, price, rating },
+  } = useSortFilter();
```

Comment:
you can destructure the state on the context itself and pass value here. code looks clean.

Reply:
Okay, I will look into it.